src/components/Table/Table.qa.stories.tsx

```diff
@@ -0,0 +1,332 @@
+  tradeCategories: string[];
+  location: string;
+  date: string;
+  priceInCents: number;
+};
+type BeamRow = SimpleHeaderAndDataWith<BeamData>;
+
+export function Fixed() {
+  return (
+    <GridTable<BeamRow>
+      style={beamFixedStyle}
+      sorting={{ on: "client", initial: [1, "ASC"] }}
+      columns={beamStyleColumns()}
+      rows={[
+        { kind: "header", id: "header" },
+        ...zeroTo(20).map((idx) => ({
+          kind: "data" as const,
+          id: `r:${idx + 1}`,
+          data: {
+            id: `r:${idx + 1}`,
+            favorite: idx < 5,
+            commitmentName: idx === 9 ? "A longer name that will truncate with an ellipsis" : `Commitment ${idx + 1}`,
+            date: `01/${idx + 1 > 9 ? idx + 1 : `0${idx + 1}`}/2020`,
+            status: "Success",
+            tradeCategories: ["Roofing", "Architecture", "Plumbing"],
+            priceInCents: 1234_56 + idx,
+            location: idx % 2 ? "l:1" : "l:2",
+          },
+        })),
+      ]}
+    />
+  );
+}
+
+export function Flexible() {
+  return (
+    <GridTable<BeamRow>
+      style={beamFlexibleStyle}
+      sorting={{ on: "client", initial: [1, "ASC"] }}
+      columns={beamStyleColumns()}
+      rows={[
+        { kind: "header", id: "header" },
+        ...zeroTo(20).map((idx) => ({
+          kind: "data" as const,
+          id: `r:${idx + 1}`,
+          data: {
+            id: `r:${idx + 1}`,
+            favorite: idx < 5,
+            commitmentName: idx === 9 ? "A longer name that will truncate with an ellipsis" : `Commitment ${idx + 1}`,
+            date: `01/${idx + 1 > 9 ? idx + 1 : `0${idx + 1}`}/2020`,
+            status: "Success",
+            tradeCategories: ["Roofing", "Architecture", "Plumbing"],
+            priceInCents: 1234_56 + idx,
+            location: idx % 2 ? "l:1" : "l:2",
+          },
+        })),
+      ]}
+    />
+  );
+}
+
+type BeamBudgetData = {
+  name?: ReactNode;
+  original?: number;
+  changeOrders?: number;
+  reallocations?: number;
+  revised?: number;
+  committed?: number;
+  difference?: number;
+  actuals?: number;
+  projected?: number;
+  costToComplete?: number;
+  children?: BeamChildRow[];
+};
+type HeaderRow = { kind: "header" };
+type BeamTotalsRow = { kind: "totals"; id: string } & BeamBudgetData;
+type BeamParentRow = { kind: "parent"; id: string } & BeamBudgetData;
+type BeamChildRow = { kind: "child"; id: string } & BeamBudgetData;
+type BeamNestedRow = BeamTotalsRow | HeaderRow | BeamParentRow | BeamChildRow;
+
+export function Nested() {
+  const totalsRowStyles = useMemo(() => ({ totals: { cellCss: beamTotalsRowStyle } }), []);
+  const rowStyles = useMemo(() => ({ parent: { cellCss: beamGroupRowStyle } }), []);
+  return (
+    <div css={Css.mw("fit-content").$}>
+      <GridTable<BeamNestedRow>
+        style={beamFixedStyle}
+        columns={beamNestedColumns}
+        rows={beamTotalsRows}
+        rowStyles={totalsRowStyles}
+      />
+      <GridTable<BeamNestedRow>
```

Comment:
Using two "GridTables" to get the Totals bar on top and ensure it lines up with the Table below. They use the same column definitions, but a different set of `row` and rowStyles. 
Perhaps at some point we can teach the GridTable component about a non-header row being above the "header" kind, if it becomes a common pattern.
For now, wrapping the two tables in a `min-width: fit-content" keeps the columns aligned between the two.

Reply:
Yeah, or maybe a new `TableWithTotals` component that has `rows` / `columns` props and then two `totalStyles` / `rowStyles` or something like that. And it'd just make two GridTables like you have here, with the wrapper.